server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -558,6 +559,41 @@ private IndexMetadata getIndexMetadata(String clusterName, String clusterUUID, U
         }
     }
 
+    /**
+     * Fetch global metadata from remote cluster state
+     *
+     * @param clusterUUID uuid of cluster state to refer to in remote
+     * @param clusterName name of the cluster
+     * @return {@link IndexMetadata}
+     */
+    public Metadata getGlobalMetadata(String clusterName, String clusterUUID) {
+        start();
+        Optional<ClusterMetadataManifest> clusterMetadataManifest = getLatestClusterMetadataManifest(clusterName, clusterUUID);
+        if (!clusterMetadataManifest.isPresent()) {
+            throw new IllegalStateException("Latest index metadata is not present for the provided clusterUUID");
+        }
+        String globalMetadataFileName = clusterMetadataManifest.get().getGlobalMetadataFileName();
+        try {
+            String[] splitPath = globalMetadataFileName.split("/");
+            return BlobStoreRepository.GLOBAL_METADATA_FORMAT.read(
+                globalMetadataContainer(clusterName, clusterUUID),
+                splitPath[splitPath.length - 1],
```

Comment:
we might not need this, can you verify once? looking at the upload PR you should able to directly use `globalMetadataFileName`

Reply:
No, in manifest file we are storing whole path, hence we need to split it and parse the file name out of it.